src/App.tsx

```diff
@@ -48,64 +52,61 @@ const App = () => {
   const pendingTxHashes = useAppSelector((s) => selectAddressesPendingTransactions(s, addressHashes)).map(
     (tx) => tx.address.hash
   )
-  const [settings, network, addressesStatus] = useAppSelector((s) => [
+  const [settings, network, addressesStatus, isPassphraseUsed] = useAppSelector((s) => [
     s.settings,
     s.network,
     s.addresses.status,
-    s.app.loading
+    s.app.loading,
+    s.activeWallet.isPassphraseUsed
   ])
 
   const [splashScreenVisible, setSplashScreenVisible] = useState(true)
   const [isUpdateWalletModalVisible, setUpdateWalletModalVisible] = useState(!!newVersion)
 
   useEffect(() => {
-    const localStorageSettings = migrateDeprecatedSettings()
+    const localStorageGeneralSettings = migrateGeneralSettings()
+    const localStorageNetworkSettings = migrateNetworkSettings()
 
-    dispatch(generalSettingsMigrated(localStorageSettings.general))
-    dispatch(networkSettingsMigrated(localStorageSettings.network))
+    dispatch(generalSettingsMigrated(localStorageGeneralSettings))
+    dispatch(networkSettingsMigrated(localStorageNetworkSettings))
 
-    migrateWalletData()
-  }, [dispatch])
+    if (!isPassphraseUsed) migrateWalletData()
+  }, [dispatch, isPassphraseUsed])
 
   const initializeClient = useCallback(async () => {
     try {
       await client.init(network.settings.nodeHost, network.settings.explorerApiHost)
       dispatch(apiClientInitSucceeded())
-      setSnackbarMessage({
-        text: `${t('Current network')}: ${network.name}.`,
-        type: 'info',
-        duration: 4000
-      })
     } catch (e) {
       dispatch(apiClientInitFailed())
-      console.error('Could not connect to network: ', network.name)
-      console.error(e)
+      console.error('Could not connect to network: ', network.name, e)
     }
-  }, [network.settings.nodeHost, network.settings.explorerApiHost, network.name, dispatch, setSnackbarMessage, t])
+  }, [network.settings.nodeHost, network.settings.explorerApiHost, network.name, dispatch])
 
   // Is there a better way to trigger the initial client initialization?
-  // Currently we trigger it "magically" by setting the the networkSlice status status to 'connecting', which is
-  // happening when loading the stored network settings and when network settings are updated by the user.
+  // Currently we trigger it is a side-effect of setting the networkSlice status to 'connecting', which is happening
+  // when loading the stored network settings and when network settings are updated by the user.
```

Comment:
On second thought, I don't think we have implemented it in a bad way.

1. The user changes the network settings
2. Redux stores the new settings and updates the network status to `connecting`
3. Our app picks up this change and tries to connect to the new network settings
   1. When it succeeds, it informs Redux that it managed to connect and Redux updates the status to `online`
   2. When it fails, it informs Redux that it failed, and Redux updates the status to `offline`
   3. The wallet then reacts to this changes accordingly, displaying a success of failure message

Reply:
I'm okay with this as well. Side-effects are "bad" when the chain of actions is murky and we spot some edge cases that cause issues. It's not the case here. Let's keep it as it is rn until we see some problems rise :)